### DISCUSSION TITLE
background_tasks block request when middleware added

### DISCUSSION THREAD

**Discussion by pheman** [fastapi/fastapi#10335] [Questions] [answered]
### First Check

- [x] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
import uvicorn
from time import sleep
from fastapi import FastAPI, Request,BackgroundTasks
from fastapi.middleware.cors import CORSMiddleware

def add_middleware(app: FastAPI):
    @app.middleware("http")
    async def add_mid(request: Request, call_next):
        response = await call_next(request)
        return response

def sleep_and_print(ns):
    for s in range(ns):
        print(f"sleeping {s}th second")
        sleep(1)


app = FastAPI()
app.add_middleware(
    CORSMiddleware,
    allow_origins=["*"],
    allow_credentials=True,
    allow_methods=["*"],
    allow_headers=["*"]
)

add_middleware(app)

@app.get("/sleep_ns")
def sleep_ns(ns: int,
           background_tasks: BackgroundTasks):
    background_tasks.add_task(sleep_and_print, ns)
    return f"sleep for {ns}s"

@app.get("/test")
def test():
    return "returned"

if __name__ == "__main__":

    uvicorn.run(
        "main:app",
        host="0.0.0.0",
        reload=False,
        port=8009
    )
```

### Description

With middleware function, the background task sleep_and_print will block the following ONE request until the background_task finishes. If the add_middleware is commented(comment Line27, add_middleware(app) ), no blocking any more.

Besides, if switch the parameter ‘Connection’ in the headers from 'keep-alive'  to 'close', there is no blocking even with  add_middleware .

Besides, the blocking seems only to occur after the following ONE request. During the blocking period, the third request will return immediately.

Above tests are on Postman. If using a Python scripts to send the requests, no blockings at all, since there are no Connection parameter in the header.

### Operating System

Linux

### FastAPI Version

0.99.1

### Pydantic Version

1.10.12

### Python Version

Python 3.10.12

**Comment by pheman** [comment]
Same with  fastapi-0.103.1.

**Comment by Kludex** [comment, marked as answer]
This is fixed on latest Starlette. FastAPI needs to bump it.

**Reply by pheman** [reply]
Is there a time schedule?

**Reply by Kludex** [reply]
I expect it to be in the next weeks, but I'm not sure. Sorry.

**Reply by pheman** [reply]
thanks!

**Reply by amacfie-tc** [reply]
see https://github.com/tiangolo/fastapi/pull/9636 and `https://github.com/tiangolo/fastapi/issues/10370#:~:text=parsing%20(increasing%20performance).-,Upgrade%20Starlette,-.`
